Kernel/Net/NetworkAdapter.cpp

```diff
@@ -199,10 +138,49 @@ size_t NetworkAdapter::dequeue_packet(u8* buffer, size_t buffer_size, Time& pack
     size_t packet_size = packet_buffer.size();
     VERIFY(packet_size <= buffer_size);
     memcpy(buffer, packet_buffer.data(), packet_size);
-    m_unused_packets.append(*packet_with_timestamp);
+    release_packet_buffer(*packet_with_timestamp);
     return packet_size;
 }
 
+RefPtr<PacketWithTimestamp> NetworkAdapter::acquire_packet_buffer(size_t size)
+{
+    InterruptDisabler disabler;
+    if (m_unused_packets.is_empty()) {
+        auto buffer = KBuffer::create_with_size(size, Region::Access::Read | Region::Access::Write, "Packet Buffer", AllocationStrategy::AllocateNow);
+        auto packet = adopt_ref_if_nonnull(new PacketWithTimestamp { move(buffer), kgettimeofday() });
+        packet->buffer.set_size(size);
```

Comment:
This could used to handle allocation failure by the looks of it, how it immediately de-references a potential null pointer.

Reply:
Same thing below on line 163.

Reply:
Fixed and also added missing checks in `{UDP,TCP,IPv4}Socket.cpp`. :+1: